### PR TITLE
Add community repos

### LINKED ISSUE #224: Support more modern PHP versions out of the box.
This will probably require some extra channels and what not, ( e.g. [using this PPA on Ubuntu for 7.2](https://launchpad.net/~ondrej/+archive/ubuntu/php) ). But I think it's worth it.

### PR DESCRIPTION
**Pull request by jeffbyrnes**
## Description

Add ability to use community-supported repositories (PPA for Debian systems, custom RPM for Red Hat/Amazon) to install more recent versions of PHP.

## Issues Resolved

Fixes #224 

## Check List

- [x] All tests pass. See TESTING.md for details.
- [x] New functionality includes testing.
- [x] New functionality has been documented in the README if applicable.

### CODE REVIEW

**Comment by kitchen-porter** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please include a CHANGELOG entry.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: README.md [hunk 1]
```diff
@@ -54,6 +54,17 @@ Include the default recipe in a run list, to get `php`. By default `php` is inst
 
 This recipe installs PHP from packages.
 
+### community_package
+
+This recipe intalls PHP from one of two available community package repositories, depending on platform family. This provides the ability to install PHP versions that are no provided by the official distro repositories.
+
+Set `node['php']['install_method'] = community_package` to use these repositories.
```
Comment (ramereth) [inline README.md:61]: Should this be the following instead?
```suggestion
Set `node['php']['install_method'] = 'community_package'` to use these repositories.
```

File: README.md [hunk 2]
```diff
@@ -54,6 +54,17 @@ Include the default recipe in a run list, to get `php`. By default `php` is inst
...
+
+Please see `test/cookbooks/test/recipes/community.rb` for an example of how to use attributes to install the desired version of PHP & its supporting packages, and please refer to the documentation on these community repositories:
+
+- RHEL/Amazon - [Remi’s RPM repository](https://rpms.remirepo.net)
```
Comment (ramereth) [inline README.md:65]: Do we want to care about IUS repos for RHEL? They currently only work on RHEL7. We use it FWIW but we may switch to Remi's at some point.

File: test/integration/resource_community/inspec/php_spec.rb [hunk 3]
```diff
@@ -0,0 +1,30 @@
+describe 'PHP' do
+  it 'has php' do
+    expect(command('php -v').exit_status).to eq(0)
```
Comment (ramereth) [inline test/integration/resource_community/inspec/php_spec.rb:3]: IMHO you should check what major version of php is installed so you know you got the correct one.

File: README.md (hunk 2 above)
Comment (jeffbyrnes) [reply README.md:65]: @ramereth I picked Remi’s b/c it appeared to be on the same level of “use this” from various search results as the APT PPA, is still kept up-to-date, and offers a broad selection of PHP versions for various flavors of EL7 & EL8.

I didn't come across the IUS repos in my search! Looking through the [IUS repo](https://repo.ius.io), seems its not as up-to-date as Remi’s; it doesn’t currently offer PHP >7.4. Pretty cool project though.

**Comment by ramereth** [comment]
@jeffbyrnes do you mind addressing the `md-links` test that's failing? It seems to be unrelated to your change but it would be nice to fix.

**Comment by jeffbyrnes** [comment]
@ramereth so here’s the thing: I tried to! But for some reason, those links always 503. If you visit the current ones in a browser, you’ll 301 → updated link, but even if I update the links to the result of the 301, md-links still says they're busted.

Maybe php.net has some kind of anti-robot setup that 503s non-human visitors?

**Comment by ramereth** [comment]
> @ramereth so here’s the thing: I tried to! But for some reason, those links always 503. If you visit the current ones in a browser, you’ll 301 → updated link, but even if I update the links to the result of the 301, md-links still says they're busted.
> 
> Maybe php.net has some kind of anti-robot setup that 503s non-human visitors?

Yeah, I wonder if they do. I'd vote to remove that specific check as I don't think we have it turned on globally. @sous-chefs/board thoughts? We mostly just need the MDL check.

**Comment by jeffbyrnes** [comment]
@ramereth well, other than md-links, things are looking green!

**Comment by ramereth** [comment]
@jeffbyrnes go ahead and remove that check in this PR so this will completely pass.

**Comment by jeffbyrnes** [comment]
Not sure why your commit didn’t show up @ramereth. I amended it slightly & force-pushed, it's here now.

**Comment by ramereth** [comment]
@jeffbyrnes yeah, GH being strange.. thanks

**Review by ramereth — APPROVED** [review]

**Comment by kitchen-porter** [comment]
Released as: [8.1.0](https://github.com/sous-chefs/php/releases/tag/8.1.0)

**Pull request merged**